### PR TITLE
chore(comparison): add webscene bookmarks support

### PR DESCRIPTION
**Pull request by dryabinin-actionengine**
Types are added here - https://github.com/visgl/loaders.gl/pull/2299

### CODE REVIEW

**Review by belom88 — CHANGES_REQUESTED** [review]

File: src/components/comparison/comparison-side/comparison-side.tsx [hunk 1]
```diff
@@ -454,15 +457,34 @@ export const ComparisonSide = ({
...
+  /**
+   * We should keep application layers structure in bookmarks. 
+   * @param newExamples 
+   * @param bookmarks 
+   */
+  const updateBookmarksWithExamples = (newExamples: LayerExample[], bookmarks: Bookmark[]) => {
+    for (const bookmark of bookmarks) {
+      bookmark.layersLeftSide = newExamples
+    }
+  }
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/components/comparison/comparison-side/comparison-side.tsx]: Removed

File: src/components/comparison/comparison-side/comparison-side.tsx [hunk 2]
```diff
@@ -454,15 +457,34 @@ export const ComparisonSide = ({
...
   };
 
+  /**
+   * We should keep application layers structure in bookmarks. 
+   * @param newExamples 
+   * @param bookmarks 
+   */
+  const updateBookmarksWithExamples = (newExamples: LayerExample[], bookmarks: Bookmark[]) => {
+    for (const bookmark of bookmarks) {
+      bookmark.layersLeftSide = newExamples
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/components/comparison/comparison-side/comparison-side.tsx]: Done

File: src/utils/bookmarks-utils.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
+import type { ArcGisWebScene, /*ArcGisCamera*/ } from "@loaders.gl/i3s/src/types";
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/utils/bookmarks-utils.ts]: Done

File: src/utils/bookmarks-utils.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+          activeLayersIdsLeftSide: supportedVisibleLayersIds,
+          layersRightSide: [],
+          activeLayersIdsRightSide: [],
+        };
+
+        bookmarks.push(bookmark);
+      }
+    }
+  } catch (error) {
+    console.log("Can't load bookmarks from webscene: ", error);
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/utils/bookmarks-utils.ts]: Done

File: src/utils/bookmarks-utils.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  return bookmarks;
+}
+
+/**
+ * Do conversion from ArcGIS spatial reference coordinates of camera to bookmark wiew state with WGS84 CRS.
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/utils/bookmarks-utils.ts]: Done

File: src/utils/bookmarks-utils.ts [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+ * It can be any CRS system in ArcGIS, but we support conversion only EPSG:3857 for now.
+ * Spec - https://developers.arcgis.com/web-scene-specification/objects/spatialReference
+ * TODO Add camera type from loaders.gl when types will be published.
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/utils/bookmarks-utils.ts]: Done

**Review by belom88 — APPROVED** [review]

**Pull request merged**